test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightPropertySerializerTests.cs

```diff
@@ -577,15 +579,55 @@ public void WritingPropertyShouldWriteCollectionResourceInstanceAnnotation()
                 result);
         }
 
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public void WritingJsonElementProperties_ShouldSerializeJsonInput()
+        {
+            string jsonInputString = "{\"foo\":\"bar\"}";
+            var jsonDocument = System.Text.Json.JsonDocument.Parse(jsonInputString);
+
+            var property = new ODataProperty()
+            {
+                Name = "JsonProp",
+                Value = new ODataJsonElementValue(jsonDocument.RootElement)
+            };
+
+            var result = SerializeProperty(null, property);
+            Assert.Equal("{\"JsonProp\":{\"foo\":\"bar\"}}", result);
+        }
+
+        [Fact]
+        public void WritingJsonElementProperties_ShouldSerializeJsonInput_WithODataUtf8JsonWriter()
+        {
+            string jsonInputString = "{\"foo\":\"bar\"}";
+            var jsonDocument = System.Text.Json.JsonDocument.Parse(jsonInputString);
+
+            var property = new ODataProperty()
+            {
+                Name = "JsonProp",
+                Value = new ODataJsonElementValue(jsonDocument.RootElement)
+            };
+
+            Action<IContainerBuilder> configureServices = (IContainerBuilder builder) =>
+            {
+                builder.AddService<IStreamBasedJsonWriterFactory>(ServiceLifetime.Singleton, _ => DefaultStreamBasedJsonWriterFactory.Default);
+            };
+
+            var result = SerializeProperty(null, property, configureServices);
+            Assert.Equal("{\"JsonProp\":{\"foo\":\"bar\"}}", result);
+        }

```

Comment:
How come the input string is not similar to the output string?

Reply:
Like to access the json object you have to do `result.JsonProp`? is the `result.JsonProp` what gets returned?

Reply:
It's because the `SerializerProperty` helper methods looks like this:

```c#
jsonLightOutputContext.JsonWriter.StartObjectScope();
serializer.WriteProperties(
    owningType,
    new[] { odataProperty },
    /*isComplexValue*/ false,
    new NullDuplicatePropertyNameChecker(),
    null);
jsonLightOutputContext.JsonWriter.EndObjectScope();
```

So, the `JsonElement` is treated as the value of the `ODataProperty` with name `JsonProp` and the property itself is wrapped inside a JSON object (delimited by the `StartObjectScope()` and `EndObjectScope()` calls);